### PR TITLE
docker: correctly set auth configs on ImageBuild

### PR DESCRIPTION
**Pull request by nicks**
Hello @maiamcc,

Please review the following commits I made in branch nicks/ch1453/private:

99087a43b42d6827f9dcff973d7ddc74c3033130 (2019-01-31 18:39:33 -0500)
docker: correctly set auth configs on ImageBuild

### CODE REVIEW

**Comment by nicks** [comment]
fixes https://github.com/windmilleng/tilt/issues/1076

File: internal/docker/client.go [hunk 1]
```diff
@@ -210,11 +222,35 @@ func NegotiateAPIVersion(ctx context.Context) func(client *client.Client) error
...
+	// https://github.com/docker/cli/blob/23446275646041f9b598d64c51be24d5d0e49376/cli/command/image/build.go#L386
+	authConfigs, _ := configFile.GetAllCredentials()
+
+	c.authConfigs = authConfigs
+
+	close(c.initDone)
+}
+
 func (c *Cli) ImageBuild(ctx context.Context, buildContext io.Reader, options types.ImageBuildOptions) (types.ImageBuildResponse, error) {
+	<-c.initDone
```
Comment (maiamcc) [inline internal/docker/client.go]: are we worried about getting stuck here? do we want a timeout + error?

**Comment by nicks** [comment]
ugh, testing this manually, it works ok on the legacy build engine but not on buildkit, this will get fussy

**Comment by nicks** [comment]
@maiamcc ok, did another pass on this, including a revamp of how we pass Options around. PTAL?

**Pull request merged**